pd/src/main.rs

```diff
@@ -214,14 +214,21 @@ async fn main() -> anyhow::Result<()> {
                     req.create_span()
                 }))
                 .service(tower_actor::Actor::new(10, |queue: _| {
-                    pd::Consensus::new(storage.clone(), queue).run()
+                    let storage = storage.clone();
+                    async move {
+                        pd::Consensus::new(storage.clone(), queue)
+                            .await?
+                            .run()
+                            .await
+                    }
                 }));
             let mempool = tower::ServiceBuilder::new()
                 .layer(request_span::layer(|req: &MempoolRequest| {
                     req.create_span()
                 }))
                 .service(tower_actor::Actor::new(10, |queue: _| {
-                    pd::Mempool::new(storage.clone(), queue).run()
+                    let storage = storage.clone();
+                    async move { pd::Mempool::new(storage.clone(), queue).await?.run().await }
```

Comment:
Can we avoid propagating this noisy sequence onto API users?

Reply:
https://github.com/penumbra-zone/penumbra/pull/2345/files/c06530db6d049b770a7d70104667e308e509e4b6#r1163369559